src/tools/w3dviewer/scaledialog.cpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * @file
+ *
+ * @author Jonathan Wilson
+ *
+ * @brief W3D View Scale Dialog
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "scaledialog.h"
+#include "resource.h"
+
+// clang-format off
+BEGIN_MESSAGE_MAP(CScaleDialog, CDialog)
+END_MESSAGE_MAP()
+// clang-format on
+
+CScaleDialog::CScaleDialog(float scale, CWnd *pParentWnd, const char *label) :
+    CDialog(IDD_SCALE, pParentWnd), m_scale(scale), m_label(label)
+{
+}
+
+BOOL CScaleDialog::OnNotify(WPARAM wParam, LPARAM lParam, LRESULT *pResult)
+{
+    return CDialog::OnNotify(wParam, lParam, pResult);
+}
+
+BOOL CScaleDialog::OnInitDialog()
+{
+    CDialog::OnInitDialog();
+    m_scaleSpin.SetRange(0, (short)75536);
```

Comment:
That value is very unusual for a short.
0xffff = 65535

Reply:
This is exactly what the original code is doing, I just copied it.